fastify.js

```diff
@@ -88,8 +88,13 @@ function fastify (options) {
 
   validateBodyLimitOption(options.bodyLimit)
 
+  const caseSensitiveParse = (string) => {
+    const shouldLower = options.caseSensitive === false
+    return querystring.parse(shouldLower ? string.toLowerCase() : string)
+  }
```

Comment:
Can you please move this to a top level function?

Reply:
@mcollina I'm trying to do a unit test this way, but I get a timeout error, could you guide me what's going here? I made if off the `custom-querystring-parser.test.js`

```
// default-querystring-parser.test.js
'use strict'

const t = require('tap')
const test = t.test
const sget = require('simple-get').concat
const Fastify = require('..')

test('Default querystring parser should keep case sensitive keys', t => {
  t.plan(9)

  const fastify = Fastify()

  fastify.get('/', (req, reply) => {
    t.deepEqual(req.query, {
      Foo: 'bar',
      Baz: 'faz'
    })
    reply.send({ hello: 'world' })
  })

  fastify.listen(0, (err, address) => {
    t.error(err)
    t.tearDown(() => fastify.close())

    sget({
      method: 'GET',
      url: `${address}?Foo=bar&Baz=faz`
    }, (err, response, body) => {
      t.error(err)
      t.strictEqual(response.statusCode, 200)
    })

    fastify.inject({
      method: 'GET',
      url: `${address}?Foo=bar&Baz=faz`
    }, (err, response, body) => {
      t.error(err)
      t.strictEqual(response.statusCode, 200)
    })
  })
})
```

Reply:
I think the use of `address` is wrong with `sget`. That;s not a `url`.

Reply:
Would you be able to put an example for me? I don't get how testing for apis work.

Here are some cases I have in mind:

- *`options.caseSensitive = true | undefined`:* `foo=bar&Foo=bar` => `{foo: 'bar', Foo: 'bar'}`
- *`options.caseSensitive = false`:* `foo=bar&Baz=bar` => `{foo: 'bar', baz: 'bar'}`
- *`options.caseSensitive = false`:* `foo=bar&Foo=Bar` => `{foo: ['bar','Bar']}`
- *`options.caseSensitive = true | undefined`:* `foo=bar&Foo=bar` => `{foo: 'bar', Foo: 'bar'}`

Reply:
nvm I found some examples, tho I'm having some issues with other tests, any chance you could look at @mcollina ?